app/models/subscription.rb

```diff
@@ -36,6 +36,13 @@ class Subscription < ApplicationRecord
             .to_h
         }
 
+  scope :dedup_by_subscriber,
+        lambda {
+          select("DISTINCT ON (subscriptions.subscriber_id) subscriber_id, subscriptions.id as id")
+            .order("subscriptions.subscriber_id DESC, subscriptions.created_at desc")
+            .map(&:id)
```

Comment:
This seems to create actual models only to throw them away with the map. I wonder if this would be better as:

```
order("subscriptions.subscriber_id DESC, subscriptions.created_at desc")
.pluck("DISTINCT ON (subscriptions.subscriber_id) subscriber_id, subscriptions.id as id")
.map(&:last)
```

Reply:
Oh yes, that's what I was aiming for - totally missed that "pluck" can have arbitrary SQL. I've amended the first commit to do that - it actually required `Arel.sql` (see updated commit message).